orchestrator-salt/src/main/resources/salt/salt/postgresql/init.sls

```diff
@@ -24,14 +26,29 @@ init-services-db-remote:
 
 init-db-with-utf8:
   cmd.run:
-    - name: rm -rf /var/lib/pgsql/data && runuser -l postgres sh -c 'initdb --locale=en_US.UTF-8 /var/lib/pgsql/data > /var/lib/pgsql/initdb.log' && rm /var/log/pgsql_listen_address_configured
+    - name: rm -rf {{ postgres_data_directory }} && runuser -l postgres sh -c 'initdb --locale=en_US.UTF-8 {{ postgres_data_directory }} > /var/lib/pgsql/initdb.log' && rm /var/log/pgsql_listen_address_configured
     - unless: grep -q UTF-8 /var/lib/pgsql/initdb.log
 
+{%- if postgres_data_on_attached_disk %}
+
+change-db-location:
+  file.replace:
+    - name: /usr/lib/systemd/system/postgresql-10.service
+    - pattern: "Environment=PGDATA=.*"
+    - repl: Environment=PGDATA=/hadoopfs/fs1/pgsql/data
```

Comment:
On Ycloud we don't have any disks at all. In that case our generation will result in /hadoopfs/root1 instance of /hadoopfs/fs1 so assuming 1 might not going to work.

Reply:
this repl value is wrong, the unless checks for the directory calculated on java side while this is hard coded: `{{ postgres_data_directory }}`

Reply:
It was just a copy-paste error.